File: LinkCity/ViewControllers/LCSplashVC.m
//
//  LCSplashVC.m
//  LinkCity
//
//  Created by roy on 3/25/15.
//  Copyright (c) 2015 linkcity. All rights reserved.
//

#import "LCSplashVC.h"
#import "LCUMengHelper.h"

@interface LCSplashVC ()

@property (weak, nonatomic) IBOutlet UIImageView *logoImageView;
@property (weak, nonatomic) IBOutlet NSLayoutConstraint *logoTop;
@property (weak, nonatomic) IBOutlet NSLayoutConstraint *logoWidth;
@property (weak, nonatomic) IBOutlet NSLayoutConstraint *logoHeight;

@property (nonatomic, strong) UIImageView *coverImageView;
@property (weak, nonatomic) IBOutlet UIView *tapView;


@property (weak, nonatomic) IBOutlet UIButton *userDeclarationButton;

@end

@implementation LCSplashVC

+ (instancetype)createInstance{
    return (LCSplashVC *)[LCStoryboardManager viewControllerWithFileName:SBNameMain identifier:VCIDSplashVC];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    if (IS_IPHONE_4_4S || IS_IPHONE_5_5S) {
        self.logoImageView.image = [UIImage imageNamed:@"SplashDuckrLogo"];
        self.logoTop.constant = 69;
        self.logoWidth.constant = 106;
        self.logoHeight.constant = 106;
    }else{
        self.logoImageView.image = [UIImage imageNamed:@"SplashDuckrLogoMin"];
        self.logoTop.constant = 110;
        self.logoWidth.constant = 118;
        self.logoHeight.constant = 118;
    }
    
    UILongPressGestureRecognizer *longPressGesture = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(longPressGestureAction:)];
    longPressGesture.minimumPressDuration = 3;
    [self.tapView addGestureRecognizer:longPressGesture];
    
    NSDictionary *attribute = @{NSFontAttributeName:[UIFont fontWithName:FONT_LANTINGBLACK size:11],
                                NSUnderlineStyleAttributeName:[NSNumber numberWithInteger:NSUnderlineStyleSingle],
                                NSForegroundColorAttributeName:UIColorFromR_G_B_A(206, 141, 54, 1)};
    NSMutableAttributedString *userDeclarationText = [[NSMutableAttributedString alloc] initWithString:@"使用条款和隐私政策" attributes:attribute];
    [self.userDeclarationButton setAttributedTitle:userDeclarationText forState:UIControlStateNormal];
}

- (void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
}

- (IBAction)registerButtonAction:(id)sender {
    [[LCUMengHelper sharedInstance] setup];
    
    AppDelegate *appDelegate = [LCSharedFuncUtil getAppDelegate];
    [appDelegate showTabBarVC];
}
- (IBAction)skipButtonAction:(id)sender {
    [[LCUMengHelper sharedInstance] setup];
    
    [LCDataManager sharedInstance].isFirstTimeOpenApp = NO;
    [[LCDataManager sharedInstance] saveData];
    
    AppDelegate *appDelegate = [LCSharedFuncUtil getAppDelegate];
    [appDelegate showTabBarVC];
}
- (void)longPressGestureAction:(UILongPressGestureRecognizer *)gestureRecognizer{
    //长按则相当于按跳过按钮
    //并永远关闭本App的友盟统计
    [LCDataManager sharedInstance].shouldUMengActive = 0;
    [[LCDataManager sharedInstance] saveData];
    
    [LCNetRequester didBlockUMengWithCallBack:^(NSError *error) {
        LCLogInfo(@"did block UMent with error: %@",error);
    }];
    
    
    [LCDataManager sharedInstance].isFirstTimeOpenApp = NO;
    [[LCDataManager sharedInstance] saveData];
    
    AppDelegate *appDelegate = [LCSharedFuncUtil getAppDelegate];
    [appDelegate showTabBarVC];
}

- (IBAction)userDeclaration:(id)sender {
    [LCViewSwitcher presentWebVCtoShowURL:server_url(SERVER_HOST, LCUserUseAgreementURL) withTitle:@"使用条款和隐私政策"];
}



//#pragma mark Blured image cover
//- (void)showCover{
//    if (!self.coverImageView) {
//        self.coverImageView = [[UIImageView alloc] initWithFrame:self.view.bounds];
//        UIImage *screenShot = [LCImageUtil createScreenShot];
//        screenShot = [LCImageUtil blurWithCoreImage:screenShot withRect:self.coverImageView.bounds pixel:2];
//        self.coverImageView.image = screenShot;
//    }
//    [self.view addSubview:self.coverImageView];
//}
//- (void)hideCover{
//    if (self.coverImageView) {
//        [self.coverImageView removeFromSuperview];
//    }
//}
//
//#pragma mark - LCRegisterAndLoginHelperDelegate
//- (void)registerAndLoginhelperDidLogin{
//    
//    [LCDataManager sharedInstance].isFirstTimeOpenApp = NO;
//    [[LCDataManager sharedInstance] saveData];
//    
//    AppDelegate *appDelegate = [LCSharedFuncUtil getAppDelegate];
//    [appDelegate showTabBarVC];
//}
//- (void)registerAndLoginHelperDidCancel{
//}

@end

File: LinkCity/ViewControllers/LCSplashVC.swift
//
//  LCSplashVC.swift
//  LinkCity
//
//  Created by 张宗硕 on 1/10/16.
//  Copyright © 2016 linkcity. All rights reserved.
//

import UIKit

class LCSplashVC: LCAutoRefreshVC {
    
    @IBOutlet weak var termButton: UIButton!
    @IBOutlet var tapView: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.initTermButton()
        self.initGestureRecognizer()
    }
    
    override class func createInstance() -> LCSplashVC {
        return LCStoryboardManager.viewControllerWithFileName(SBNameMain, identifier: VCIDSplashVC) as! LCSplashVC
    }
    
    // MARK: Init Functions.
    internal func initTermButton() {
        let attributedString = NSMutableAttributedString(string: "使用条款和隐私政策")
        attributedString.addAttribute(NSUnderlineStyleAttributeName, value: NSUnderlineStyle.StyleDouble.rawValue, range: NSMakeRange(0, attributedString.length))
        attributedString.addAttribute(NSForegroundColorAttributeName, value: LCSSharedFuncUtil.colorFromRGBA(0xad7f2d, alpha: 1.0), range: NSMakeRange(0, attributedString.length))
        self.termButton.setAttributedTitle(attributedString, forState: .Normal)
    }
    
    internal func initGestureRecognizer() {
        let longPressGesture = UILongPressGestureRecognizer(target: self, action: #selector(LCSplashVC.longPressGestureAction))
        longPressGesture.minimumPressDuration = 3
        self.tapView.addGestureRecognizer(longPressGesture)
    }
    
    // MARK: Common Functions.
    func longPressGestureAction(gestureRecognizer: UILongPressGestureRecognizer) {
        print("did block UMent with error")
        /// 长按则相当于按跳过按钮，并永远关闭本App的友盟统计.
//        LCDataManager.sharedInstance().shouldUMengActive = 0
        LCDataManager.sharedInstance().saveData()
        LCNetRequester.didBlockUMengWithCallBack { (error: NSError?) -> Void in
            print("did block UMent with error: %@", error)
        }
        
        let appDelegate = LCSharedFuncUtil.getAppDelegate()
        appDelegate.showTabBarVC()
    }
    
    // MARK: UIButton Functions.
    @IBAction func registerButtonAction(sender: AnyObject) {
//        LCUMengHelper.sharedInstance().setup()
        
        let appDelegate = LCSharedFuncUtil.getAppDelegate()
        appDelegate.showTabBarVC()
    }
    
    @IBAction func loginButtonAction(sender: AnyObject) {
//        LCUMengHelper.sharedInstance().setup()
        
        LCDataManager.sharedInstance().isFirstInUseLogin = true
        let appDelegate = LCSharedFuncUtil.getAppDelegate()
        appDelegate.showTabBarVC()
    }
    
    @IBAction func tryButtonAction(sender: AnyObject) {
        LCDataManager.sharedInstance().isFirstTimeOpenApp = 0
        LCDataManager.sharedInstance().saveData()
        
//        LCUMengHelper.sharedInstance().setup()
        
        let appDelegate = LCSharedFuncUtil.getAppDelegate()
        appDelegate.showTabBarVC()
    }
    
    @IBAction func termButtonAction(sender: AnyObject) {
        LCViewSwitcher.presentWebVCtoShowURL(LCSSharedFuncUtil.serverUrl(LCConstants.serverHost(), suffix:LCUserUseAgreementURL), withTitle: "使用条款和隐私策略")
    }
}
